pkg/nmstate/policy.go

```diff
@@ -320,6 +321,71 @@ func (builder *PolicyBuilder) WithVlanInterface(baseInterface string, vlanID uin
 	return builder.withInterface(newInterface)
 }
 
+// WithVlanInterfaceIP adds VLAN interface with IP configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithVlanInterfaceIP(baseInterface string, vlanID uint16,
+	ipv4Addresses, ipv6Addresses string) *PolicyBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with VLAN interface %s and vlanID %d",
+		builder.Definition.Name, baseInterface, vlanID)
+
+	if baseInterface == "" {
+		glog.V(100).Infof("The baseInterface can not be empty string")
+
+		builder.errorMsg = "nodenetworkconfigurationpolicy 'baseInterface' cannot be empty"
+	}
```

Comment:
```suggestion
	if baseInterface == "" {
		glog.V(100).Infof("The baseInterface can not be empty string")

		builder.errorMsg = "nodenetworkconfigurationpolicy 'baseInterface' cannot be empty"
		
		return builder
	}
```
For new code we should follow this pattern and then we don't need the
```go
if builder.errorMsg != "" {
        return builder
}
```

Reply:
removed